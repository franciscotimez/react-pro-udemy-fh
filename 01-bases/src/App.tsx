import { Counter } from "./bases/Counter";
import { CounterBy } from "./bases/CounterBy";
import { CounterEffect } from "./bases/CounterEffect";
import { CounterHook } from "./bases/CounterHook";

function App() {
  return (
    <>
      <h1>React</h1>
      <hr/>
      <CounterHook />
      <CounterEffect />
      <CounterBy initialValue={0}/>
      <Counter initialValue={0}/>
    </>
  );
}

export default App;
