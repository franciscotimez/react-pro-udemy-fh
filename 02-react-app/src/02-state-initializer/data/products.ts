import { Product } from "../interfaces/interfaces";

export const products: Product[] = [
  {
    id: "1",
    title: "Coffe Mug",
    img: './coffee-mug.png'
  },
  {
    id: "2",
    title: "Coffe Mug - Meme",
    img: './coffee-mug2.png'
  }
];