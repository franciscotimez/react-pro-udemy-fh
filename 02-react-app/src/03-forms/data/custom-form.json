[
  {
    "type": "input",
    "name": "firstName",
    "placeholder": "Francisco",
    "label": "First Name",
    "value": "Juan",
    "validations": [
      {
        "type": "required"
      }
    ]
  },
  {
    "type": "input",
    "name": "lastName",
    "placeholder": "Timez",
    "label": "Last Name",
    "value": "Ramirez",
    "validations": [
      {
        "type": "required"
      },
      {
        "type": "minLength",
        "value": 5
      }
    ]
  },
  {
    "type": "email",
    "name": "email",
    "placeholder": "example@example.com",
    "label": "Email",
    "value": "",
    "validations": [
      {
        "type": "required"
      },
      {
        "type": "email"
      }
    ]
  },
  {
    "type": "select",
    "name": "favoriteGame",
    "label": "Favorite Game",
    "value": "",
    "options": [
      {
        "id": "1",
        "label": "Select an Option"
      },
      {
        "id": "2",
        "label": "Super Smash"
      },
      {
        "id": "3",
        "label": "Metal Gear"
      },
      {
        "id": "4",
        "label": "Resident Evil"
      }
    ]
  }
]