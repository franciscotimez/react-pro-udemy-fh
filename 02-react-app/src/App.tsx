import { Navigation } from "./routes/Navigation";

export const App = () => {
  return (
    <>
      <Navigation />
    </>
  )
}

