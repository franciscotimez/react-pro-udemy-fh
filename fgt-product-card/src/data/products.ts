import { Product } from "../interfaces/interfaces";

export const product1: Product = {
  id: "1",
  title: "Coffe Mug - Sin imagen",
  // img: './coffee-mug.png'
};

export const product2: Product = {
  id: "1",
  title: "Coffe Mug - Con imagen",
  img: './coffee-mug.png'
};