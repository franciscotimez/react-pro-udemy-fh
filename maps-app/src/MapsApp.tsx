import { PlacesProvider } from "./context";

export const MapsApp = () => {
  return (
    <PlacesProvider>
      <h1>Hola mundo de nuevo!</h1>
    </PlacesProvider>
  )
}
