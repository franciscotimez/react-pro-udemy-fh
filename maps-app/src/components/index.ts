export { SearchBar } from './SearchBar';

export { ReactLogo } from './ReactLogo';
export { BtnMyLocation } from './BtnMyLocation';
export { Loading } from "./Loading";
export { MapView } from "./MapView";
