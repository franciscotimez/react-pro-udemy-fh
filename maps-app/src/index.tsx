import React from 'react';
import ReactDOM from 'react-dom/client';

import { MapsApp } from './MapsApp';

import mapboxgl from 'mapbox-gl'; // or "const mapboxgl = require('mapbox-gl');"
 
mapboxgl.accessToken = 'pk.eyJ1IjoiZnJhbnRpbWV6IiwiYSI6ImNsY2dqM3Y5cDNka3kzcGw3bG01b3I0encifQ.Ti2e1awq7MT8PLLkI-goSg';

if (!navigator.geolocation) {
  alert('Geo location is missing');
  throw new Error("Geolocation is missing");
}

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <MapsApp />
  </React.StrictMode>
);
