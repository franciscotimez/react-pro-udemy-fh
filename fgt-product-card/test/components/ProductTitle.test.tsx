import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductTitle } from '../../src';
import { product1 } from '../../src/data/products';


describe('Pruebas en ProductTitle', () => {
  test('Debe de mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom Product" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
