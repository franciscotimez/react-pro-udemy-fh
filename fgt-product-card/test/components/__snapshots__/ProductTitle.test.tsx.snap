// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductTitle Debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - Sin imagen
  </span>
</div>
`;

exports[`Pruebas en ProductTitle Debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
